piptools/writer.py

```diff
@@ -137,7 +137,7 @@ def _format_requirement(self, ireq, reverse_dependencies, primary_packages, mark
             for hash_ in sorted(ireq_hashes):
                 line += " \\\n    --hash={}".format(hash_)
 
-        if not self.annotate or ireq.name in primary_packages:
+        if not self.annotate or ireq.name.lower() in primary_packages:
```

Comment:
Good catch
While at it, as `primary_packages` was generated using `key_from_req`, we should use that here too for consistency.
`key_from_req(ireq.req) in primary_packages`